src/core/meshcop/dataset_local.hpp

```diff
@@ -54,7 +54,7 @@ class DatasetLocal : public InstanceLocator
      * @param[in]  aType      The type of the dataset, active or pending.
      *
      */
-    DatasetLocal(Instance &aInstance, Tlv::Type aType);
```

Comment:
I think we should not add `const` and remove `const` from the corresponding `*.cpp` files.